src/coreclr/hosts/coreconsole/coreconsole.cpp

```diff
@@ -44,8 +44,8 @@ class StringBuffer {
             m_buffer = new wchar_t[m_defaultSize];
             m_capacity = m_defaultSize;
         }
-        if (m_length + strLen + 1 > m_capacity) {
-            size_t newCapacity = m_capacity * 2;
+        while (m_length + strLen + 1 > m_capacity) {
```

Comment:
Can you please explain why while is needed if the new capacity is going to be computed to be greater than the existing m_capacity?

And why the factor of 1.5 vs 2?